cmd/clusterctl/cmd/delete.go

```diff
@@ -17,15 +17,82 @@ limitations under the License.
 package cmd
 
 import (
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+type deleteOptions struct {
+	kubeconfig           string
+	forceDeleteNamespace bool
+	forceDeleteCRD       bool
+	deleteAll            bool
+}
+
+var dd = &deleteOptions{}
+
 var deleteCmd = &cobra.Command{
-	Use:   "delete",
-	Short: "Delete a cluster API resource",
-	Long:  `Delete a cluster API resource with one command`,
+	Use:   "delete [providers]",
+	Short: "Deletes one or more providers from the management cluster",
+	Long: LongDesc(`
+		Deletes one or more providers from the management cluster.`),
+
+	Example: Examples(`
+		# Deletes the AWS provider
+		# Please note that this imply the deletion of everything all the provider components except the 
```

Comment:
"Please note that this implies the deletion of all provider components except the hosting namespace and the CRDs"

Reply:
Also wondering why `clusterctl delete aws` can't also delete the hosting namespace and CRD in the end? Is it because a user may end up deploying additional resources into that namespace?

Reply:
The rationale behind this is to avoid incidental deletion of any user-created object unless explicitly requested to do so. Given that the user might have created other objects in the namespace, IMO deleting namespace should not happen by default

Reply:
I feel slightly different here -- as a starting point I think we should not allow custom namespaces for provider components and also assume a user is not going to add to `cap*-system`. We should warn users against this if they are adding things to `cap*-system`

Reply:
>  I think we should not allow custom namespaces for provider components 

This boils up to the discussion about multi-tenancy, that unfortunately is not addressed yet (see https://github.com/kubernetes-sigs/cluster-api/issues/1631). ATM this can't be rules out